src/storage/patch/RdfImmutableCheckPatcher.ts

```diff
@@ -0,0 +1,75 @@
+import type { Quad } from 'rdf-js';
+import type { AuxiliaryStrategy } from '../../http/auxiliary/AuxiliaryStrategy';
+import { BasicRepresentation } from '../../http/representation/BasicRepresentation';
+import type { Representation } from '../../http/representation/Representation';
+import { getLoggerFor } from '../../logging/LogUtil';
+import { INTERNAL_QUADS } from '../../util/ContentTypes';
+import { ConflictHttpError } from '../../util/errors/ConflictHttpError';
+import { serializeQuads, uniqueQuads } from '../../util/QuadUtil';
+import { cloneRepresentation } from '../../util/ResourceUtil';
+import { readableToQuads, readableToString } from '../../util/StreamUtil';
+import { RDF, PIM, LDP } from '../../util/Vocabularies';
+import type { RepresentationPatcherInput } from './RepresentationPatcher';
+import { RepresentationPatcher } from './RepresentationPatcher';
+
+export class RdfImmutableCheckPatcher extends RepresentationPatcher {
+  protected readonly logger = getLoggerFor(this);
+
+  private readonly patcher: RepresentationPatcher;
+  private readonly metaStrategy: AuxiliaryStrategy;
+
+  public constructor(patcher: RepresentationPatcher, metaStrategy: AuxiliaryStrategy) {
+    super();
+    this.patcher = patcher;
+    this.metaStrategy = metaStrategy;
+  }
+
+  public async handle(input: RepresentationPatcherInput): Promise<Representation> {
+    let { representation } = input;
+    const { identifier } = input;
+
+    representation = representation ?? new BasicRepresentation([], INTERNAL_QUADS);
+    const immutableTriples: [string, string | undefined][] = [];
+    immutableTriples.push([ RDF.type, PIM.Storage ]);
+    immutableTriples.push([ LDP.contains, undefined ]);
+
+    let patched: Representation;
+    if (this.metaStrategy.isAuxiliaryIdentifier(identifier)) {
+      const clonedRepresentation = await cloneRepresentation(representation);
+      patched = await this.patcher.handleSafe(input);
+      const clonedPatchedRepresentation = await cloneRepresentation(patched);
```

Comment:
I want to see a solution with no `cloneRepresentation` calls. There are a few cases in the code where there is no way around cloning but this is not one of them in my opinion.

My suggestion would be to change `N3Patcher` and `SparqlUpdatePatcher` so the take as input an N3 store instead of a `Representation`, and also output such a store instead. That new interface can be called `RdfStorePatcher`.

A new class `RdfPatcher` would then be needed that drains the incoming `Representation` into an N3 store, feeds it into the source `RdfStorePatcher`, and converts the resulting store back into a representation.

The `RdfImmutableCheckPatcher` would then also be an instance of the `RdfStorePatcher`. It can then easily check the triples in the incoming store and in the output store without having to duplicate a stream multiple times.